lte/gateway/c/session_manager/PipelinedClient.cpp

```diff
@@ -64,21 +63,59 @@ magma::DeactivateFlowsRequest create_deactivate_req(
   req.set_uplink_tunnel(teids.agw_teid());
   req.set_remove_default_drop_flows(remove_default_drop_rules);
   req.mutable_request_origin()->set_type(origin_type);
-  auto mut_versioned_rules = req.mutable_policies();
+  return req;
+}
+
+/**
+ * @brief Create a map of Teids -> DeactivateFlowsRequest
+ * If to_process is empty, create one default_teids -> DeactivateFlowsRequest
+ * with no rules. For each to_process item, create item.teid ->
+ * DeactivateFlowsRequest with item.rules
+ * @param imsi
+ * @param ip_addr
+ * @param ipv6_addr
+ * @param default_teids this value is only used if to_process is empty
+ * @param to_process
+ * @param origin_type
+ * @param remove_default_drop_rules
+ * @return magma::DeactivateReqByTeids
+ */
+magma::DeactivateReqByTeids make_deactivate_req_by_teid(
+    const std::string& imsi, const std::string& ip_addr,
+    const std::string& ipv6_addr, const magma::Teids default_teids,
+    const magma::RulesToProcess& to_process,
+    const magma::RequestOriginType_OriginType origin_type,
+    const bool remove_default_drop_rules) {
+  magma::DeactivateReqByTeids deactivate_req_by_teids;
+  if (to_process.empty()) {
+    // Send an empty request with the default teid
+    deactivate_req_by_teids[default_teids] = make_deactivate_req(
+        imsi, ip_addr, ipv6_addr, default_teids, origin_type,
+        remove_default_drop_rules);
+    return deactivate_req_by_teids;
```

Comment:
will this option deactivate all default and non default bearers. 
meaming, will pipelined look for all the flows related to that imsi, no matter teid?. Shouldn't you include any dedicated teid?

Reply:
no for removing all default and non default bearers, we have to specify the specific teids. (PipelineD doesn't have a concept of default vs dedicated, unfortunately)
I just included the clause to stay consistent with the current behavior. But in reality, for termination we use `deactivate_flows_for_rules_for_termination`, which takes in a vector of teids so we can make a request for all of them.